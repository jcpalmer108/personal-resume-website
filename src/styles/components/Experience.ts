import styled from "styled-components"
import { silver_polish } from "../variables"

export const Wrapper = styled.div`
  text-align: center;
`

export const Upper = styled.div`

`

export const Title = styled.h2`
  padding-bottom: 30px;
  width: 75%;
  margin: auto;
`

export const Description = styled.div`
  font-family: Gilroy-Regular;

`

export const Paragraph = styled.p`
  font-size: 16px;
  line-spacing: 26px;
  margin: 0px;
  padding-bottom: 16px;

  &:first-child {
    padding-bottom: 0px;
  }
`

type TileProps = {
  desktop?: boolean | undefined
}

export const Mosaic = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  column-gap: 25px;
`
export const Tile = styled.a<TileProps>`
  display: ${props => props.desktop ? 'none' : 'flex'};
  flex-direction: column;
  align-items: center;
  justify-content: end;

  border: 1px solid ${silver_polish};
  text-decoration: none;
  color: ${silver_polish};
  padding-bottom: 24px;
  margin-top: 45px;
  height: 148px;
  width: 175px;
  row-gap: 35px;
`

export const Icon = styled.img`
  height: 45px;
  width: 45px;

`

export const Label = styled.span`
  font-family: Gilroy-Medium;
  font-size: 13px;
  text-transform: uppercase;

`

export const Lower = styled.div`

`

export const Job = styled.div`
  display: flex;
  flex-direction: column;
  padding-top: 30px;

`

export const Logo = styled.img`
  padding: 20px;
  width: 100px;
  height: 100px;
  margin: auto;
`

export const Content = styled.div`
  display: flex;
  flex-direction: column;

`

export const JobTitle = styled.div`
  font-family: Butler-Regular;
  font-size: 25px;
  line-spacing: 65px;
`

export const Info = styled.span`
  font-family: Gilroy-Regular;
  font-style: italic;
  font-size: 14px;
  padding: 12px 12px 20px 12px;
`

export const JobDescription = styled.div`
  font-family: Gilroy-Regular;

`

export const JobParagraph = styled.p`
  font-size: 16px;
  line-spacing: 26px;
  margin: 0px;
  padding-bottom: 16px;

  &:first-child {
    padding-bottom: 0px;
  }




`

export const DiagonalLine = styled.img`
  display: none;
`
