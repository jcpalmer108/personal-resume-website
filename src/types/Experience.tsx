import { Sections } from "./content"

export type ExperienceProps = {
  content: Sections | undefined
}