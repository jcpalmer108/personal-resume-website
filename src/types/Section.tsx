import { ReactNode } from 'react'

export type SectionProps = {
  children: ReactNode,
  label?: string,
  light?: boolean,
  thin?: boolean,
  wide?: boolean,
  center?: boolean,
  bottom?: boolean,
  noTop?: boolean,
  noBorder?: boolean
}

export type WrapperProps = {
  light?: boolean,
  thin? : boolean,
  bottom?: boolean,
  noBorder?: boolean
}

export type HeaderProps = {
  center?: boolean,
  noTop?: boolean
}

export type ChildrenProps = {
  padded?: boolean
}
