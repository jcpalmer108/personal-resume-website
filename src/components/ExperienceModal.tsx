import { Experience } from "../types/content"
import { 
  Content,
  Circle,
  Logo,
  Tile,
  Overview,
  TileLabel,
  TileInfo,
  OrgTeam,
  JobDetails,
  Summary
} from "../styles/components/Modal";
import Details from "./Details"
import Project from "./Project"
import Modal from "./Modal"

type ExperienceModalProps = {
  closeModal: Function
  content?: Experience;
}

export default function ExperienceModal({ closeModal, content }: ExperienceModalProps) {
  const generateDetails = () => {
    let response = [];

    if(content?.description) response.push(
      <Details 
        key="Description"
        data-testid="Description" 
        label="Description" 
        info={content.description}
      />
    )

    if(content?.organization && content?.team) response.push(
      <OrgTeam key="OrgTeam">
        <Details 
          label="Organization" 
          info={[content.organization]} 
          data-testid="Organization" 
        />
        <Details 
          label="Team" 
          info={[content.team]} 
          noBottom 
          data-testid="Team" 
        />
      </OrgTeam>
    )

    if(content?.location) response.push(
      <Details 
        key="Location"
        label="Location" 
        info={[content.location]}
        data-testid="Location" 
      />
    )

    if(content?.timeline) response.push(
      <Details 
        key="Timeline"
        label="Timeline" 
        info={[`${content.timeline.start} - ${content.timeline.end}`]} 
        data-testid="Timeline" 
      />
    )
    
    if(content?.projects) {
      response.push(<Details key="Projects" label="Projects" data-testid="Projects" />)
      response.push(
      <div data-testid="ProjectLineItems">
        {
          content.projects.map(({icon, industry, description, title}, index) => 
            <Project 
              key={`Project ${index + 1}`}
              icon={icon} 
              industry={industry} 
              description={description} 
              title={title} 
              data-testid={`Project ${index + 1}`}
            />
          )
        }
      </div>)
    }

    return response
  }

  return (
    <>
      <Modal closeModal={closeModal}>
        <Content data-testid="Content">
          <Tile>
            <Circle>
              <Logo data-testid="Logo" src={require(`../assets/images/${content?.icon || "nike"}-white.svg`)} alt="logo" icon={content?.icon} />
            </Circle>
            <Overview>
              <Summary>
                <TileLabel>Title</TileLabel>
                <TileInfo data-testid="Title">{content?.title}</TileInfo>
              </Summary>
              <Summary>
                <TileLabel>Employer</TileLabel>
                <TileInfo data-testid="Employer">{content?.employer}</TileInfo>
              </Summary>
            </Overview>
          </Tile>
          <JobDetails>
            {generateDetails().map((item, index) => <div key={`Detail ${index + 1}`}>{item}</div>)}
          </JobDetails>
        </Content>
      </Modal>
    </>
  )
}