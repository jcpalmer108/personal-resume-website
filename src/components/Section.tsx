import { 
  Wrapper, 
  Header, 
  Line, 
  Label, 
  Title, 
  Children
} from "../styles/components/Section";
import line from "../assets/images/label-line.svg";
import { SectionProps } from "../types/Section"

export default function Section({ 
  label,
  children,
  light,
  thin,
  wide,
  center,
  bottom,
  noTop,
  noBorder
}: SectionProps) {
  return (
    <Wrapper noBorder={noBorder} light={light} thin={thin} bottom={bottom} data-testid="Wrapper">
      {label && (
        <Header noTop={noTop} data-testid="Header" center={center} >
          <Label data-testid="Label">
            <Line src={line} alt="test"/>
            <Title>{label.toUpperCase()}</Title>
          </Label>
        </Header>
      )}
      <Children padded={!wide} data-testid="Children">{children}</Children>
    </Wrapper>
  )
}