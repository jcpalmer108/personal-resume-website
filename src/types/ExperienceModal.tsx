import { Experience } from "../types/content"

export type ExperienceModalProps = {
  closeModal: Function
  content?: Experience;
}
