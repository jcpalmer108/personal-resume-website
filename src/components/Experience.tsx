import { useState } from "react";
import callMe from "../assets/images/icons/call-me.svg";
import diagonal from "../assets/images/icons/diagonal-line.svg";
import emailMe from "../assets/images/icons/email-me.svg";
import {
  Content,
  Description,
  DiagonalLine,
  Icon,
  Info,
  Job,
  JobDescription,
  JobParagraph,
  JobTitle,
  Label,
  Logo,
  Lower,
  ModalTrigger,
  Mosaic,
  Paragraph,
  Summary,
  Tile,
  Title,
  Upper,
  Wrapper,
} from "../styles/components/Experience";
import { ExperienceProps } from "../types/Experience";
import ExperienceModal from "./ExperienceModal";
import Section from "./Section";

const defaultModalContent = {
  title: "",
  employer: "",
  location: "",
  description: [""],
  icon: "",
  timeline: {
    start: "",
    end: "",
  },
};

export default function Experience({ content, contact }: ExperienceProps) {
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const [selectedContent, setSelectedContent] = useState(
    content?.subSection?.experience
      ? content?.subSection?.experience[0]
      : defaultModalContent
  );

  const manageSelectedContent = (selectedIndex: number) => {
    if (
      content &&
      content.subSection &&
      content.subSection.experience &&
      content.subSection.experience[selectedIndex]
    ) {
      setSelectedContent(content.subSection.experience[selectedIndex]);
    } else if (
      content &&
      content.subSection &&
      content.subSection.experience &&
      content.subSection.experience[0]
    ) {
      setSelectedContent(content.subSection.experience[0]);
    } else {
      setSelectedContent(defaultModalContent);
    }
  };

  if (!content || !contact) {
    return null;
  }

  return (
    <div data-testid="Experience" id="experience">
      {modalIsOpen && (
        <ExperienceModal
          closeModal={() => setModalIsOpen(!modalIsOpen)}
          content={selectedContent}
        />
      )}
      <Section center label={content?.label}>
        <Wrapper>
          <Upper>
            <Summary>
              <Title data-testid="Title">{content?.title}</Title>
              <Description data-testid="Description">
                {content?.description &&
                  content?.description.map((paragraph, index) => (
                    <Paragraph key={`Paragraph ${index + 1}`}>
                      {paragraph}
                    </Paragraph>
                  ))}
              </Description>
            </Summary>
            <Mosaic desktop>
              <Tile href={`tel:${contact?.cell}`} data-testid="Call">
                <Icon src={callMe} alt="call me" />
                <Label>CALL ME</Label>
              </Tile>
              <Tile href={`mailto:${contact?.email}`} data-testid="Email">
                <Icon src={emailMe} alt="email me" />
                <Label>EMAIL ME</Label>
              </Tile>
            </Mosaic>
          </Upper>
          <Lower>
            {content?.subSection?.experience &&
              content?.subSection?.experience.map((job, index) => (
                <Job
                  key={`Job ${index + 1}`}
                  onClick={() => {
                    manageSelectedContent(index);
                    setModalIsOpen(!modalIsOpen);
                  }}
                >
                  <Logo
                    src={require("../assets/images/company/" +
                      job.icon +
                      ".svg")}
                    alt={job.icon}
                  />
                  <Content>
                    <JobTitle data-testid={`Title ${index + 1}`}>
                      {job.title}
                    </JobTitle>
                    <Info data-testid={`Info ${index + 1}`}>
                      {job.employer} / {job.timeline.start} - {job.timeline.end}
                    </Info>
                  </Content>
                  <DiagonalLine src={diagonal} alt="forward slash" />
                  <JobDescription data-testid={`Details ${index + 1}`}>
                    <JobParagraph key={`Paragraph ${index + 1}`}>
                      {job.description[0]}
                      <ModalTrigger>... Read More</ModalTrigger>
                    </JobParagraph>
                  </JobDescription>
                </Job>
              ))}
            <Mosaic>
              <Tile href={`tel:${contact?.cell}`}>
                <Icon src={callMe} alt="call me" />
                <Label>Call Me</Label>
              </Tile>
              <Tile href={`mailto:${contact?.email}`}>
                <Icon src={emailMe} alt="email me" />
                <Label>Email Me</Label>
              </Tile>
            </Mosaic>
          </Lower>
        </Wrapper>
      </Section>
    </div>
  );
}
