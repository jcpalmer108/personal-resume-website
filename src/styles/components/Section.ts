import styled from "styled-components";
import { desktopBreakpoint, contentWidth, silver_polish, tabletBreakpoint } from "../variables";
import { WrapperProps, HeaderProps, ChildrenProps } from "../../types/Section"

export const Wrapper = styled.div<WrapperProps>`
  max-width: ${contentWidth}px;
  margin: auto;
  border-left: solid;
  border-right: solid;
  border-bottom: solid;
  border-color: transparent;
  border-width: 0px;

  @media (min-width: ${contentWidth}px) {
    ${props => props.noBorder ? `
      border: none;
    ` : `
      border-bottom: ${props.bottom ? "soild" : "none" };
      border-color: ${props.light ? "white" : silver_polish};
      border-width: ${props.thin ? 0.2 : 1}px;    
    `}
  }
`

export const Header = styled.div<HeaderProps>`
  width: fit-content;
  margin: ${props => props.center ? "auto" : "0px"};
  padding-top: 50px; 
  padding-left: 16px;
  
  @media (min-width: ${tabletBreakpoint}px) {
    margin: 0px;
    padding-left: 25px;
    padding-top: ${props => props.noTop ? '25px' : '80px'};
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-left: 60px;
    padding-top: 85px;
  }
`

export const Label = styled.div`
  display: flex;
  column-gap: 10px;
  font-size: 11px;
  font-family: Gilroy-Regular;

  @media (min-width: ${tabletBreakpoint}px) {
    transform: rotate(-90deg);
    padding-bottom: calc(100% - 15px);
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding-bottom: calc(100% - 17px);
    font-size: 16px;
  }

`

export const Children = styled.div<ChildrenProps>`
  padding: ${props => props.padded ? "20px 15px" : "0px"};

  @media (min-width: ${tabletBreakpoint}px) {
    padding: ${props => props.padded ? "40px 25px" : "0px"};
  }

  @media (min-width: ${desktopBreakpoint}px) {
    padding: ${props => props.padded ? "45px 60px 55px 60px" : "0px"};
  }
`


export const Title = styled.span`
  font-size: 12px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
  }

`

export const Line = styled.img`
  width: 25px;
`

