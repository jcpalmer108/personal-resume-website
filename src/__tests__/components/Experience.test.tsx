import { SectionProps } from "../../types/Section";
import { render, screen } from "@testing-library/react";
import Experience from "../../components/Experience";
import { ExperienceModalProps } from "../../types/ExperienceModal";

// mock components
const mockSection = jest.fn();
jest.mock(
  "../../components/Section",
  () =>
    ({
      label,
      children,
      light,
      thin,
      wide,
      center,
      bottom,
      noTop,
      noBorder,
    }: SectionProps) => {
      mockSection(label, light, thin, wide, center, bottom, noTop, noBorder);
      return <div>{children}</div>;
    }
);

const mockExperienceModal = jest.fn();
jest.mock(
  "../../components/ExperienceModal",
  () =>
    ({ closeModal, content }: ExperienceModalProps) => {
      mockExperienceModal(closeModal, content);
      return <div />;
    }
);

// params needed for component
const params = {
  required: {
    content: {
      key: "experience",
      label: "Experience",
      title: "Test title",
      description: ["test description"],
      inMenu: true,
      subSection: {
        experience: [
          {
            title: "test title 1",
            employer: "test Company 1",
            location: "test location 1",
            description: ["test desc 1"],
            icon: "test",
            timeline: {
              start: "test start 1",
              end: "test end 1",
            },
            projects: [
              {
                industry: "industry 1",
                title: "title project 1",
                description: ["desc 1"],
                icon: "test",
              },
              {
                industry: "industry 2",
                title: "title project 2",
                description: ["desc 2"],
                icon: "test",
              },
            ],
          },
          {
            title: "test title 2",
            employer: "test company 2",
            team: "test team",
            organization: "test organization",
            location: "test location 2",
            description: ["test desc 2"],
            icon: "test",
            timeline: {
              start: "test start 1",
              end: "test end 2",
            },
          },
        ],
      },
    },
    contact: {
      email: "test@email.com",
      cell: "1112223333",
    },
  },
};

describe("Experience", () => {
  test("renders if only required params are passed in", () => {
    // given
    const { content, contact } = params.required;
    render(<Experience content={content} contact={contact} />);

    //then
    expect(screen.getByTestId("Experience")).toMatchSnapshot();
    expect(screen.getByTestId("Experience")).toHaveAttribute(
      "id",
      "experience"
    );
    expect(screen.getByTestId("Title")).toHaveTextContent(content.title);
    content.description.forEach((paragraph, index) => {
      expect(
        screen.getByTestId("Description").childNodes[index]
      ).toHaveTextContent(paragraph);
    });
    expect(screen.getByTestId("Call")).toBeTruthy();
    expect(screen.getByTestId("Call")).toHaveAttribute(
      "href",
      `tel:${contact.cell}`
    );
    expect(screen.getByTestId("Email")).toBeTruthy();
    expect(screen.getByTestId("Email")).toHaveAttribute(
      "href",
      `mailto:${contact.email}`
    );

    content.subSection.experience.forEach((job, index) => {
      expect(screen.getByTestId(`Title ${index + 1}`)).toHaveTextContent(
        job.title
      );
      expect(screen.getByTestId(`Info ${index + 1}`)).toHaveTextContent(
        job.employer + " / " + job.timeline.start + " - " + job.timeline.end
      );
      expect(screen.getByTestId(`Details ${index + 1}`)).toHaveTextContent(
        job.description[0] + "... Read More"
      );
      expect(screen.getAllByRole("img", { name: job.icon })).toHaveLength(2);
    });
  });

  test("does not render if content or contact is not defined", () => {
    // given
    render(<Experience content={undefined} contact={undefined} />);

    // then
    expect(screen.queryByTestId("Experience")).toBeFalsy();
  });
});
