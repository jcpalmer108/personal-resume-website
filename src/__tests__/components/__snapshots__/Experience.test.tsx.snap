// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders if only required params are passed in 1`] = `
<div
  data-testid="Experience"
  id="experience"
>
  <div>
    <div
      class="sc-bczRLJ cBafEA"
    >
      <div
        class="sc-gsnTZi kXvfBR"
      >
        <div
          class="sc-dkzDqf bSZmJh"
        >
          <h2
            class="sc-hKMtZM bpteWv"
            data-testid="Title"
          >
            Test title
          </h2>
          <div
            class="sc-eCYdqJ fmEXNN"
            data-testid="Description"
          >
            <p
              class="sc-jSMfEi bCXkKM"
            >
              test description
            </p>
          </div>
        </div>
        <div
          class="sc-gKXOVf fOEKDU"
        >
          <a
            class="sc-iBkjds kyamSD"
            data-testid="Call"
            href="tel:1112223333"
          >
            <img
              alt="call me"
              class="sc-ftvSup jmuxGn"
              src="call-me.svg"
            />
            <span
              class="sc-papXJ eAMGEG"
            >
              CALL ME
            </span>
          </a>
          <a
            class="sc-iBkjds kyamSD"
            data-testid="Email"
            href="mailto:test@email.com"
          >
            <img
              alt="email me"
              class="sc-ftvSup jmuxGn"
              src="email-me.svg"
            />
            <span
              class="sc-papXJ eAMGEG"
            >
              EMAIL ME
            </span>
          </a>
        </div>
      </div>
      <div
        class="sc-jqUVSM"
      >
        <div
          class="sc-kDDrLX eFrgfC"
        >
          <img
            alt="test"
            class="sc-iqcoie iaIySc"
            src="[object Object]"
          />
          <div
            class="sc-crXcEl fEhkOw"
          >
            <div
              class="sc-evZas kEEaIb"
              data-testid="Title 1"
            >
              test title 1
            </div>
            <span
              class="sc-breuTD hpKqQa"
              data-testid="Info 1"
            >
              test Company 1
               / 
              test start 1
               - 
              test end 1
            </span>
          </div>
          <img
            alt="forward slash"
            class="sc-fEOsli jgFhpH"
            src="diagonal-line.svg"
          />
          <div
            class="sc-ksZaOG UlYvj"
            data-testid="Details 1"
          >
            <p
              class="sc-hAZoDl hxfvXd"
            >
              test desc 1
              <span
                class="sc-fnykZs ykQEb"
              >
                ... Read More
              </span>
            </p>
          </div>
        </div>
        <div
          class="sc-kDDrLX eFrgfC"
        >
          <img
            alt="test"
            class="sc-iqcoie iaIySc"
            src="[object Object]"
          />
          <div
            class="sc-crXcEl fEhkOw"
          >
            <div
              class="sc-evZas kEEaIb"
              data-testid="Title 2"
            >
              test title 2
            </div>
            <span
              class="sc-breuTD hpKqQa"
              data-testid="Info 2"
            >
              test company 2
               / 
              test start 1
               - 
              test end 2
            </span>
          </div>
          <img
            alt="forward slash"
            class="sc-fEOsli jgFhpH"
            src="diagonal-line.svg"
          />
          <div
            class="sc-ksZaOG UlYvj"
            data-testid="Details 2"
          >
            <p
              class="sc-hAZoDl hxfvXd"
            >
              test desc 2
              <span
                class="sc-fnykZs ykQEb"
              >
                ... Read More
              </span>
            </p>
          </div>
        </div>
        <div
          class="sc-gKXOVf cMYwWY"
        >
          <a
            class="sc-iBkjds kyamSD"
            href="tel:1112223333"
          >
            <img
              alt="call me"
              class="sc-ftvSup jmuxGn"
              src="call-me.svg"
            />
            <span
              class="sc-papXJ eAMGEG"
            >
              Call Me
            </span>
          </a>
          <a
            class="sc-iBkjds kyamSD"
            href="mailto:test@email.com"
          >
            <img
              alt="email me"
              class="sc-ftvSup jmuxGn"
              src="email-me.svg"
            />
            <span
              class="sc-papXJ eAMGEG"
            >
              Email Me
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
